aodndata/moorings/products_handler.py

```diff
@@ -224,6 +240,12 @@ def _log_excluded_files(self, errors):
                 else:
                     self.excluded_files[f].update(e)
 
+    def _add_to_collection(self, product_url):
+        """Add a new product file to the file_collection to be harvested and uploaded."""
+        product_file = PipelineFile(product_url, file_update_callback=self._file_update_callback)
```

Comment:
Lost the account of the number of times I see this pattern of setting a field after initialization. We should allow setting the publish_type at PipelineFile at init time too : ```PipelineFile(...,publish_type= ...HARVEST_UPLOAD)```. I've to remember to raise this in aodncore - any thoughts?

Reply:
I see your point, but that's nothing to do with this PR. It's an aodncore issue